# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/rocr-runtime/hotfix-non-x86.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- ROCR-Runtime-rocm-6.3.3/runtime/hsa-runtime/core/util/lnx/os_linux.cpp.vanilla	2025-03-25 17:25:44.334728911 +0100
+++ ROCR-Runtime-rocm-6.3.3/runtime/hsa-runtime/core/util/lnx/os_linux.cpp	2025-03-25 17:39:18.713507343 +0100
@@ -338,12 +338,14 @@
   std::vector<LibHandle> ret;
   std::vector<std::string> names;
 
+#if defined(__x86_64__) || defined(__i386__)
   /* Iterate through all of the loaded shared libraries in the process */
   dl_iterate_phdr(callback, &names);
 
   if (!names.empty()) {
     for (auto& name : names) ret.push_back(LoadLib(name));
   }
+#endif
 
   return ret;
 }
